src/main/java/edu/cornell/kfs/coa/document/CuAccountGlobalMaintainableImpl.java

```diff
@@ -106,22 +106,22 @@ private Stream<MaintenanceLock> generateTrickleDownMaintenanceLocks(Account inac
      */
     @Override
     public void saveBusinessObject() {
```

Comment:
@cah292 Looking at the history of our customization I think we did the override to fix downstream inactivations (KFSPTS-25409). I think we need to add the super call in our override. I don't think we intentionally want to avoid the functionality in the super. Please let me know if you find otherwise.

Reply:
It looks like we had overridden this to be similar to the total override in `ObjectCodeGlobalMaintainableImpl`, so that the downstream BO inactivation can be dealt with while iterating over the saves to the Accounts. Our total override already duplicates much of the inherited global maintainable logic; thus, simply inserting a call to the superclass's variant could result in the Account changes being saved twice (and might also interfere with the downstream BO inactivation).

If you want this method to explicitly call the superclass one instead of doing a total override, then it needs to be modified to use a different strategy for determining which Accounts were inactivated by the document. Perhaps it could be modified to invoke our custom `getExistingAccountClosedStatuses()` method a second time after the superclass has finished saving the BO changes, though more testing is needed to confirm.

If you think such rework is too involved or too complex to handle at this time, then I would suggest copying the update-role-assignments logic into our total override, to keep the processing consistent with the base code changes. After that, we could use a follow-up user story to potentially simplify the method override so that it can explicitly invoke the superclass's variant instead.

Please let me know your thoughts on which strategy would make more sense.

Reply:
I see what you mean, I will update.